commercial-schedules/src/main/java/org/dcsa/conformance/standards/cs/party/CsSubscriber.java

```diff
@@ -96,13 +99,17 @@ private void getPointToPointRoutings(JsonNode actionPrompt) {
     SuppliedScenarioParameters ssp =
         SuppliedScenarioParameters.fromJson(actionPrompt.get("suppliedScenarioParameters"));
 
+    Map<String, Collection<String>> queryParams = ssp.getMap().entrySet().stream()
+      .collect(
+        Collectors.toMap(
+          entry -> entry.getKey().getQueryParamName(),
+          entry -> Set.of(entry.getValue())));
+    if (actionPrompt.has("cursor")) {
+      queryParams.put("cursor", List.of(actionPrompt.get("cursor").asText()));
+    }
```

Comment:
I see 3 new blocks of code, with (almost?) identical code blocks. Please extract method and reuse your code constructs where it makes sense (or comment here if it is not a good idea etc).

Reply:
Nicely improved! 👍🏻